common/persistence/namespaceReplicationQueue.go

```diff
@@ -223,25 +324,15 @@ func (q *namespaceReplicationQueueImpl) GetMessagesFromDLQ(
 func (q *namespaceReplicationQueueImpl) UpdateDLQAckLevel(
 	lastProcessedMessageID int64,
 ) error {
-
-	if err := q.queue.UpdateDLQAckLevel(
-		lastProcessedMessageID,
-		localNamespaceReplicationCluster,
-	); err != nil {
-		return err
-	}
-
-	q.metricsClient.Scope(
-		metrics.PersistenceNamespaceReplicationQueueScope,
-	).UpdateGauge(
-		metrics.NamespaceReplicationDLQAckLevelGauge,
-		float64(lastProcessedMessageID),
-	)
-	return nil
+	return q.updateAckLevelWithRetry(lastProcessedMessageID, localNamespaceReplicationCluster, true)
 }
 
 func (q *namespaceReplicationQueueImpl) GetDLQAckLevel() (int64, error) {
-	dlqMetadata, err := q.queue.GetDLQAckLevels()
+	metadata, err := q.queue.GetDLQAckLevels()
+	if err != nil {
+		return EmptyQueueMessageID, err
+	}
```

Comment:
```go
switch err.(type) {
case *serviceerror.NotFound:
    return EmptyQueueMessageID, nil
default:
    return nil, err
}

Reply:
and make queue impl return serviceerror.NotFound when there is no record?

Reply:
This method returns a int64, cannot use nil. The error from store layer should already handle the case.